File: Job39_PHP Script to Execute EXE with Parameters and Display Result/C++/Sum.cpp
#include <iostream>
#include <fstream>
#include <cstdlib>

using namespace std;

int main(int argc, char* argv[]) {
    // Kiểm tra xem có đúng hai đối số được cung cấp không
    if (argc != 3) {
        cout << "Usage: sum.exe <num1> <num2>" << endl;
        return 1;
    }

    // Chuyển đổi hai đối số thành số nguyên
    int num1 = atoi(argv[1]);
    int num2 = atoi(argv[2]);

    // Kiểm tra chuyển đổi thành công
    if (num1 == 0 && argv[1][0] != '0') {
        cout << "First argument must be a valid integer." << endl;
        return 1;
    }
    if (num2 == 0 && argv[2][0] != '0') {
        cout << "Second argument must be a valid integer." << endl;
        return 1;
    }

    // Tính tổng của hai số
    int sum_result = num1 + num2;

    // Ghi kết quả vào file "result.txt"
    ofstream outfile("result.txt");
    if (!outfile) {
        cout << "Failed to open result.txt" << endl;
        return 1;
    }
    outfile << sum_result;
    outfile.close();

    return 0;
}

File: Job39_PHP Script to Execute EXE with Parameters and Display Result/python/Sum.py
import sys

if len(sys.argv) != 3:
    print("Usage: sum.py <num1> <num2>")
    sys.exit(1)

try:
    num1 = int(sys.argv[1])
    num2 = int(sys.argv[2])
except ValueError:
    print("Both arguments must be valid integers.")
    sys.exit(1)

sum_result = num1 + num2

with open("result.txt", "w") as f:
    f.write(str(sum_result))
